CDAP-492: Update Java Stream Client integration tests.

amakogoncv opened the pull request:
Updated Java Stream Client integration tests according to the issue description [CDAP-488](https://issues.cask.co/browse/CDAP-488). @anwar6953 , please take a look.

amakogoncv committed:
CDAP-492: Modified stream client java API integration tests.

amakogoncv committed:
CDAP-492: Modified stream client java API integration tests.

amakogoncv committed:
Merge branch 'develop' into CDAP-493

amakogoncv committed:
CDAP-492: Updated Stream Client integration tests.

amakogoncv committed:
CDAP-492: Updated Java Stream Client integration tests.

anwar6953 commented on cdap-integration-tests/cdap-stream-client-integration-tests/src/test/java/co/cask/cdap/client/StreamClientIT.java:
```diff
@@ -16,73 +16,164 @@
-    client.truncate(TEST_STREAM);
+    streamClient.truncate(TEST_STREAM);
+    streamEvents = streamClientTestHelper.getStreamEvents(TEST_STREAM);
+    Assert.assertEquals(0, streamEvents.size());
```
I think it will be good to check the returned contents (to ensure they match with what we sent), in addition to the checking just that the size matches.

amakogoncv replied:
will done

anwar6953 commented:
Most of the test cases are good now, but one concern:

I included a statement that should fail the integration test into one of the test cases:
`Assert.assertEquals(1, 2);`

However, when I run this command to run the test:
`mvn package -P it-local`
it still shows BUILD SUCCESS at the end.

This isn't proper, right.

amakogoncv commented:
I tried to do the same and got the same result, but I noticed that IT were not run. Please, try to run with such command: 
`mvm -rf crap-integration-tests clean install -P it-local`

For me this works ok and I got assertion error, as expected with : 
`Assert.assertEquals(1, 2);`

anwar6953 commented:
Yea, you're right regarding the integration test not running.
Turns out the test will run for 'clean install', but not for 'clean package':
`mvn -rf cdap-integration-tests clean package -P it-local`

amakogoncv committed:
CDAP-492: Added stream events validation to the test case.

anwar6953 commented:
Mostly looks good to me.

Though I have one question:
In this PR: https://github.com/caskdata/cdap-ingest/pull/94
there is a utility StreamReader. Would it be better to also use that here? Or to use the streamReader in here over there? (to avoid redundancy)

amakogoncv commented:
Yes, `StreamReader` could be reuse here.

amakogoncv committed:
Merge branch 'cdap-489' into CDAP-492

amakogoncv committed:
Merge branch 'cdap-489' into CDAP-492

amakogoncv committed:
Merge branch 'cdap-489' into CDAP-492

amakogoncv committed:
Merge branch 'cdap-489' into CDAP-492

amakogoncv committed:
CDAP-492: Updated Stream Client Tests with using cdap-common-integration-tests dependency.

amakogoncv committed:
CDAP-492: Removed unused field.

amakogoncv commented:
@anwar6953 , I have updated `StreamReader` for adapt it for all integration tests. Then I have updated Stream Client integration tests. So, for submit this PR, https://github.com/caskdata/cdap-ingest/pull/94 PR should be reviewed and submitted at first.

amakogoncv committed:
Merge branch 'develop' into CDAP-492

sreevatsanraman commented:
@amakogoncv PR 94 is merged. Please pull in the changes and do the needful here.

sreevatsanraman commented:
lgtm
